src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -147,72 +148,163 @@ handle_t halide_hexagon_remote_get_symbol(handle_t module_ptr, const char* name,
 
 volatile int power_ref_count = 0;
 
-int halide_hexagon_remote_power_hvx_on() {
-    if (power_ref_count == 0) {
-        HAP_power_response_t power_info;
+int halide_hexagon_HAP_power_hvx_on(halide_hvx_power_perf_t *perf) {
```

Comment:
I think this function should go away and it's functionality should be built into `halide_hexagon_power_hvx_on_perf`. Then, `halide_hexagon_power_hvx_on_mode` should call that function, instead of this one.

This would eliminate a round trip of struct packing/unpacking, and put all of the power_ref_count logic in one place.